node-api/handlers/beacon/handler.go

```diff
@@ -31,12 +32,11 @@ type Handler struct {
 }
 
 // NewHandler creates a new handler for the beacon API.
-func NewHandler(backend Backend) *Handler {
+func NewHandler(backend Backend, logger log.Logger) *Handler {
 	h := &Handler{
-		BaseHandler: handlers.NewBaseHandler(
-			handlers.NewRouteSet(""),
-		),
-		backend: backend,
+		BaseHandler: handlers.NewBaseHandler(logger),
+		backend:     backend,
 	}
+	registerRoutes(h)
```

Comment:
This is the other point of this  PR:

- Each handler embeds a `BaseHandler` which contains all the routes for the handler.
  - These routes are exposed to the middleware to duly direct every request
- Route registration requires passing the handler method needed to process the transaction. So there is a bit of circular dependency where:
  - BaseHandler is created when handler is created
  - Routes can be passed to the embedded base handler only after the Handler has been created

I cannot get rid of the circular dependency, but I encapsulated this into the handler constructor, rather than genereting the routing mapping ex post.